test/test_filesystem.cpp

```diff
@@ -419,9 +419,9 @@ TEST_F(TestFilesystemFixture, calculate_directory_size) {
 #ifdef WIN32
```

Comment:
I would keep the non-recursive test case like following.

- non-recursive test case: rcutils_calculate_directory_size(test/dummy_folder/dummy-subfolder/dummy-subfolder)
- recursive test case: rcutils_calculate_directory_size(test/dummy_folder)

Reply:
This modification is to support recursion.  So not support non-recursion.   
If there is non-recursive test case, do you mean 2 API  --- one for non-recursion and the other for recursion ?

Reply:
my bad, i wasn't clear enough. what i mean is not about the function but test cases,

- test only files in a single directory. (non-recursive search `test` case)
- test directories and files. (recursive search `test` case)